src/cache/http_cache.go

```diff
@@ -69,21 +73,27 @@ func (cache *httpCache) StoreExtra(target *core.BuildTarget, key []byte, file st
 	}
 }
 
-func (cache *httpCache) Retrieve(target *core.BuildTarget, key []byte) bool {
+func (cache *httpCache) Retrieve(target *core.BuildTarget, key []byte) *core.BuildMetadata {
 	// We can't tell from outside if this works or not (as we can for the dir cache)
 	// so we must assume that a target with no artifacts can't be retrieved. It's a weird
 	// case but a test already exists in the plz test suite so...
-	retrieved := false
-	for _, out := range cacheArtifacts(target) {
-		if !cache.RetrieveExtra(target, key, out) {
-			return false
+	var metadata *core.BuildMetadata
+	for _, out := range target.Outputs() {
+		if !cache.retrieveOne(target, key, out) {
+			return nil
+		}
+		metadata = &core.BuildMetadata{}
```

Comment:
Don't need to do this in the loop?

Reply:
it's covered by the comment above (it assumes that a target with no outputs can't be retrieved).

FWIW a rewrite of http_cache is coming which will fix this silliness, but want to do it after this interface change (which will make it a lot easier)